internal/configs/mock_provider.go

```diff
@@ -65,12 +72,39 @@ func decodeMockProviderBlock(block *hcl.Block) (*Provider, hcl.Diagnostics) {
 	return provider, diags
 }
 
+func extractOverrideDuring(content *hcl.BodyContent) (*string, hcl.Diagnostics) {
```

Comment:
I think you could also make this return a string without a pointer -> just return "apply" if the attribute isn't set.

Reply:
In this section [internal/configs/mock_provider.go#L317-L325](https://github.com/hashicorp/terraform/pull/36227/files#diff-e07a696d4818a4328954e8900c8e5833aeee855a5541c57c7f677e66848da9caR317-R325), the field needs to be a pointer. This is necessary because, if the `override_*` block does not explicitly set a value for `override_during`, the global value should be applied. However, if the override_* block does set a value, it should take precedence over the global value. Using a pointer allows us to differentiate between an explicitly set value and a zero value.

Reply:
ah nice! Makes sense!